process/release_process.md

```diff
@@ -0,0 +1,154 @@
+
+  - for **snd_be**, the commit ID should be used
+  - while for **xen** / **linux** / **optee-os**, the corresponding community versions or xen-troops branches should be used with **???-xt-???** naming.
+
+- Commits to **xt-core** are merged into the **'main'** branch based on evolutionary edits.
+- Additional branches in **xt-core** repositories can be temporarily created to meet the developer's needs.
+
+  **Finally, such branches should be merged into the main branch or closed!**
+- The tag-version of the **xt-core** entity changes only if the commit to a specific xt-core repository breaks the compilation with other xt-core repositories, i.e., the change requires edits in several xt-core repositories at once.
+  Example. We have the xt-core 'v2.0.0' release tag. As part of the new feature, we made some changes in meta-xt-rcar and meta-xt-common. Now, meta-xt-rcar will no longer compile with an older baseline of the meta-xt-common. In such a case, all xt-core repos receive the tag v2.1.0 after this change. So, it is clear to all clients that if they want to get some of the 'v2.1.0' xt-core features, they must switch ALL xt-core repositories at least to the 'v2,1,0' tag.
+- Suppose **xt-core** is already on version v2.0.0, and a bug is found in v2.0.0. I.e., it manifested itself in the product v8.0.0 based on v2.0.0. In such a case, the bug fix goes as a commit to the **'main'** branch of **xt-core**, and the fixes needed for a specific product go to the v.8.0.1 of the product's release branches in the form of .bbappend or patches on c/c++ code, which is changed in the baseline after the **xt-core** v2.0.0.
+
+**Note!** Changes to essential components, such as the **Xen**, may require an increment of the **xt-core** release tag.
+**Note!** Files of version history and release notes should be included in the release with a description like: *"For xt-core v2.0.0, the product should set the following variables for dom0, domD, etc. And stop using these variables - X, Y, Z, etc."*
+**Note!** Some README.md containing the current affairs is needed. The specification of all supported configuration variables the product should set for target domains, information regarding which variables have default values, etc.
+
+It leads us to need a separate 'xt-core' repository that will be the entry point into the xt-core repo system. It will not have code but documentation on which repositories and variables to set.
+
+### moulin-based products
+
+- **Approach #1**:
+  - Each release should be represented as a separate release branch.
+
+    **NO** - having a set of tags in your main branch is not enough!
+  - If you have a set of repositories released as one entity, use the same release branch name for all the related repositories.
+  - Git tags should be used, but they should be used to identify the release candidates and minor releases.

```

Comment:
Well, you can (and should) tag any release. We definetelly need branches for MAJOR releases but it is debatable if we should create branch for MINOR or PATCH release.

Reply:
Hi @lorc, I've added more specific MAJOR, MINOR, and PATCH terms:

----

- **Approach #1**:
  - Each **MAJOR** release should be represented as a separate release branch.

    **NO** - having a set of tags in your main branch is not enough!
  - If you have a set of repositories released as one entity, use the same release branch name for all the related repositories.
  - Git tags should be used, but they should be used to identify the **MINOR/PATCH** releases.
  
  ----
  
  Hopefully that would be sufficient.